### DISCUSSION TITLE
Quartz rewrite jobs after restart

### DISCUSSION THREAD

**Discussion by aunikitin** [quartznet/quartznet#991] [Q&A] [answered]
Hello.
I use Quartz.NET in .NET Core 2.2.
Quartz version is 3.2.
Npgsql version is 4.0.0 (Project has some dependencies which interfere to update)
Quartz integration looks like on the screenshot:
![Quartz_implementation](https://user-images.githubusercontent.com/38141491/96164273-6e68a580-0f34-11eb-9f7b-3ca80e125c77.jpg)
With delegate `Action<IServiceCollectionQuartzConfigurator> configure` I inject my jobs like in the screenshot below:
![Job_injection](https://user-images.githubusercontent.com/38141491/96164471-b4256e00-0f34-11eb-9e54-359fed6cb74d.jpg)
I have a simple job:
![image](https://user-images.githubusercontent.com/38141491/96164699-09617f80-0f35-11eb-9124-ce9c9cec2167.png)
I'll expect that after execution some information would be passed to persistente store (like lastSuccessExecutionTime, but in real life it should store some status) and I'll expect that my service will work like a kuber pod (that means that it can suddenly restarted).
How should I configure Quartz to store job's JobDataMap and not to rewrite it after restart.
Nowadays, I wrote crutch, that on Quartz configuring gets current job's data and store them in dictionary (see screenshot)
![image](https://user-images.githubusercontent.com/38141491/96165182-ba681a00-0f35-11eb-9a8d-0be1de169a6a.png)
and after in job initializing it injects that job data 
![image](https://user-images.githubusercontent.com/38141491/96165362-f3a08a00-0f35-11eb-87a0-e3df5463a290.png)
_____
I'll expect that after Quartz restart it doesn't rewrite jobs or something like that.

**Comment by lahma** [comment, marked as answer]
As long as you don't touch the job data map from outside it should persist between invocations and also restarts. The thing is that default configuration will overwrite jobs and triggers if they already exist (follow the XML scheduling plugin logic). The configuration is not shown in main Quartz options that configure callback gets at the moment, probably should..

You should probably configure like this:

```csharp
services.Configure<QuartzOptions>(options =>
{
    options.Scheduling.OverWriteExistingData = false;
    options.Scheduling.IgnoreDuplicates = true;
});
```
Here's the configuration description:

https://github.com/quartznet/quartznet/blob/ca3c9cd311a8c0dc57fbb5967cce94af9c66a01e/src/Quartz.Extensions.DependencyInjection/SchedulingOptions.cs#L5-L32

**Reply by lahma** [reply]
I created #992 to improve docs and API.

**Reply by aunikitin** [reply]
It works. Thanks a lot!
